community/kernel/src/main/java/org/neo4j/kernel/impl/util/dbstructure/DbStructureGuide.java

```diff
@@ -0,0 +1,254 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.util.dbstructure;
+
+import java.util.Iterator;
+
+import org.neo4j.graphdb.DependencyResolver;
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Label;
+import org.neo4j.graphdb.RelationshipType;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.helpers.ThisShouldNotHappenError;
+import org.neo4j.helpers.collection.Iterables;
+import org.neo4j.kernel.GraphDatabaseAPI;
+import org.neo4j.kernel.api.CountsRead;
+import org.neo4j.kernel.api.ReadOperations;
+import org.neo4j.kernel.api.Statement;
+import org.neo4j.kernel.api.StatementTokenNameLookup;
+import org.neo4j.kernel.api.TokenNameLookup;
+import org.neo4j.kernel.api.constraints.UniquenessConstraint;
+import org.neo4j.kernel.api.exceptions.KernelException;
+import org.neo4j.kernel.api.exceptions.LabelNotFoundKernelException;
+import org.neo4j.kernel.api.exceptions.index.IndexNotFoundKernelException;
+import org.neo4j.kernel.api.index.IndexDescriptor;
+import org.neo4j.kernel.impl.api.Kernel;
+import org.neo4j.kernel.impl.core.ThreadToStatementContextBridge;
+import org.neo4j.tooling.GlobalGraphOperations;
+
+import static java.lang.String.format;
+
+import static org.neo4j.kernel.api.CountsRead.ANY_LABEL;
+import static org.neo4j.kernel.api.CountsRead.ANY_RELATIONSHIP_TYPE;
+
+public class DbStructureGuide implements Visitable<DbStructureVisitor>
+{
+    private static Label WILDCARD_LABEL = new Label()
+    {
+        @Override
+        public String name()
+        {
+            return "";
+        }
+    };
+
+    private static RelationshipType WILDCARD_REL_TYPE = new RelationshipType()
+    {
+        @Override
+        public String name()
+        {
+            return "";
+        }
+    };
+
+    private final GraphDatabaseAPI db;
+    private final Kernel kernel;
+    private final ThreadToStatementContextBridge bridge;
+    private final GlobalGraphOperations glops;
+
+    public DbStructureGuide( GraphDatabaseService graph )
+    {
+        this.db = (GraphDatabaseAPI) graph;
+        DependencyResolver dependencyResolver = db.getDependencyResolver();
+        this.kernel = dependencyResolver.resolveDependency( Kernel.class );
+        this.bridge = dependencyResolver.resolveDependency( ThreadToStatementContextBridge.class );
+        this.glops = GlobalGraphOperations.at( db );
+    }
+
+    public void accept( DbStructureVisitor visitor )
+    {
+        try ( Transaction tx = db.beginTx() )
+        {
+            try ( Statement statement = bridge.instance() )
+            {
+                accept( statement, visitor );
+            }
+            tx.success();
+        }
+    }
+
+    private void accept( Statement statement, DbStructureVisitor visitor )
```

Comment:
I wouldn't call all of these methods "accept...", since they aren't accepting the thing the nouns in their method names. [`acceptTokens`](https://github.com/neo4j/neo4j/pull/4023/files#diff-d0b73ea01dcf1619277ef7589249b650R113) for example, does not accept any tokens, it provides the tokens to the visitor. Same goes for the other methods after here (including this `accept(...)` method). The [`accept( Visitor )`](https://github.com/neo4j/neo4j/pull/4023/files#diff-d0b73ea01dcf1619277ef7589249b650R85) method above is suitably named though, since it accepts the visitor, lets it in and allows it to visit the insides of the state.